contrib/pkg/clusterpool/clusterclaim.go

```diff
@@ -0,0 +1,100 @@
+package clusterpool
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+type ClusterClaimOptions struct {
+	Name            string
+	Namespace       string
+	Lifetime        time.Duration
+	ClusterPoolName string
```

Comment:
Regarding the confusion around logging: suggest adding a field here: "log log.FieldLogger"